File: 2025-10-03 REFATORACAO (Java)/classes/Organizador.java
package JogosVorazes.classes;

public class Organizador {
    private String nome;
    private Distrito[] distritos;

    public Organizador(String nome, Distrito[] distritos) {
        this.nome = nome;
        this.distritos = distritos;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Distrito[] getDistritos() {
        return distritos;
    }

    public void setDistritos(Distrito[] distritos) {
        this.distritos = distritos;
    }

    public String exibirDistritos() {
        String[] distritosNomes = new String[distritos.length];
        for (int i=0; i < this.distritos.length; i++) {
            distritosNomes[i] = distritos[i].getNome();
        }
        return String.join(", ", distritosNomes);
    }

    public void liberarAnimais() {
        int indexAleatorio_Distrito = (int) (Math.random() * this.distritos.length);
        System.out.println("O Organizador " + this.nome + " soltou animais no Distrito " + this.distritos[indexAleatorio_Distrito].getNome() + "! Corra!");
    }

    public void criarDesastre() {
        String[] desastres = {"Chuva Ácida", "Terremoto", "Fumaça Tóxica", "Lanças de Fogo"};
        int indexAleatorio_Desastre = (int) (Math.random() * desastres.length);
        int indexAleatorio_Distrito = (int) (Math.random() * this.distritos.length);
        System.out.println("O Organizador " + this.nome + " liberou " + desastres[indexAleatorio_Desastre] + " no Distrito " + this.distritos[indexAleatorio_Distrito].getNome() + "!");
    }
}

File: 2025-06-03 VISIBILIDADE/exercicio1a/organizador.ts
import { Distrito } from "./distrito";

export class Organizador {
    private nome: string;
    private distritos: Distrito[]

    constructor(nomeRecebido: string, distritosRecebidos: Distrito[]) {
        this.nome = nomeRecebido;
        this.distritos = distritosRecebidos;        
    }

    public setNome(nomeRecebido: string): void {
        this.nome = nomeRecebido;
    }

    public setDistritos(distritosRecebidos: Distrito[]): void {
        this.distritos = distritosRecebidos;
    }

    public getNome(): string {
        return this.nome;
    }

    public getDistritos(tipo:string= ''): any {
        if (tipo == 'nomes') {
            let distritosNomes = [];
            for (let i=0; i < this.distritos.length; i++) {
                distritosNomes.push(this.distritos[i].getNome())
            }
            return distritosNomes.join(", ")

        } else {
            return this.distritos;
        }
    
    }

    public liberarAnimais(): void {
        const distritoAleatorio = Math.floor(Math.random() * this.distritos.length);
        console.log(`O Organizador ${this.nome} soltou animais no Distrito ${this.distritos[distritoAleatorio].getNome()}! Corra!`)
    }

    public criarDesastre(): void {
        let listaDesastre = ['Chuva Ácida', 'Terremoto', 'Fumaça Tóxica', 'Lanças de Fogo']
        const desastreAleatorio = Math.floor(Math.random() * listaDesastre.length);
        const distritoAleatorio = Math.floor(Math.random() * this.distritos.length);
        console.log(`O Organizador ${this.nome} liberou o desastre ${listaDesastre[desastreAleatorio]} no Distrito ${this.distritos[distritoAleatorio].getNome()}!`)
    }
}
